profiling/src/profiling.rs

```diff
@@ -861,24 +861,34 @@ impl Profiler {
         labels: &[Label],
         locals: &RequestLocals,
     ) -> SampleMessage {
-        let mut sample_types = vec![
-            ValueType {
+        let mut sample_types = vec![];
+        let mut sample_values = vec![];
+
+        if locals.profiling_enabled {
+            sample_types.push(ValueType {
                 r#type: Cow::Borrowed("sample"),
                 unit: Cow::Borrowed("count"),
-            },
-            ValueType {
+            });
+            sample_types.push(ValueType {
                 r#type: Cow::Borrowed("wall-time"),
                 unit: Cow::Borrowed("nanoseconds"),
-            },
-            ValueType {
+            });
+            sample_values.push(samples.interrupt_count);
+            sample_values.push(samples.wall_time);
+        }
+
+        if locals.profiling_experimental_allocation_enabled {
```

Comment:
I think we need to refactor a bit: if profiling isn't enabled, why are we preparing a sample message in the first place? We shouldn't have even collected the samples in the first place, yeah?